diff --git a/src/native/corehost/apphost/static/CMakeLists.txt b/src/native/corehost/apphost/static/CMakeLists.txt
index e1a02ece4..69ac0093b 100644
--- a/src/native/corehost/apphost/static/CMakeLists.txt
+++ b/src/native/corehost/apphost/static/CMakeLists.txt
@@ -184,12 +184,16 @@ endif()
 
 if(CLR_CMAKE_TARGET_APPLE)
     LIST(APPEND NATIVE_LIBS
-        System.Security.Cryptography.Native.Apple-Static
+        System.Security.Cryptography.Native.OpenSsl-Static
+        palrt
+        coreclrpal_dac
+        corguids
+        dbgutil
     )
 
-    # Additional requirements for System.Security.Cryptography.Native.Apple
-    include(${CLR_SRC_NATIVE_DIR}/libs/System.Security.Cryptography.Native.Apple/extra_libs.cmake)
-    append_extra_cryptography_apple_libs(NATIVE_LIBS)
+    # Additional requirements for System.Security.Cryptography.Native.OpenSsl
+    include(${CLR_SRC_NATIVE_DIR}/libs/System.Security.Cryptography.Native/extra_libs.cmake)
+    append_extra_cryptography_libs(NATIVE_LIBS)
 endif()
 
 #
